codedog/utils/git_log_analyzer.py

```
@@ -0,0 +1,448 @@
+    current_diff = []
+
+    for line in result.stdout.splitlines():
+        if not line.strip():
+            continue
+
+        # Check if line starts with a file status (e.g., "M\tfile.py")
+        if line.startswith(("A\t", "M\t", "D\t")):
+            if current_file and current_diff:
+                file_diffs[current_file] = {
+                    "diff": "\n".join(current_diff),
+                    "status": current_status,
```

Ruff (0.8.2)

Undefined name `current_status`

(F821)